### PR TITLE
For Wasi build on windows, do not set CLR_CMAKE_TARGET_WIN32

### PR DESCRIPTION
**Pull request by yowl**
This PR removes `CLR_CMAKE_TARGET_WIN32` from  `eng\native\configureplatform.cmake` for the native build of WebAssembly Wasi on windows.  Follows discussion at https://github.com/dotnet/runtimelab/pull/2268#discussion_r1192802324

### CODE REVIEW

File: eng/native/configureplatform.cmake [hunk 1]
```diff
@@ -462,7 +462,9 @@ if(CLR_CMAKE_TARGET_UNIX)
         clr_unknown_arch()
     endif()
 else()
```
Comment (jkotas) [inline eng/native/configureplatform.cmake:464]:
```suggestion
endif(CLR_CMAKE_TARGET_UNIX)
```

File: eng/native/configureplatform.cmake [hunk 2]
```diff
@@ -462,7 +462,9 @@ if(CLR_CMAKE_TARGET_UNIX)
...
-    set(CLR_CMAKE_TARGET_WIN32 1)
+    if(NOT (CLR_CMAKE_TARGET_WASI))
+        set(CLR_CMAKE_TARGET_WIN32 1)
+    endif()
 endif(CLR_CMAKE_TARGET_UNIX)
```
Comment (jkotas) [inline eng/native/configureplatform.cmake:468]:
```suggestion

if(CLR_CMAKE_TARGET_OS STREQUAL windows)
      set(CLR_CMAKE_TARGET_WIN32 1)
endif()
```

File: eng/native/configureplatform.cmake [hunk 3]
```diff
@@ -461,10 +461,12 @@ if(CLR_CMAKE_TARGET_UNIX)
     else()
         clr_unknown_arch()
     endif()
-else()
-    set(CLR_CMAKE_TARGET_WIN32 1)
 endif(CLR_CMAKE_TARGET_UNIX)
 
+if(CLR_CMAKE_TARGET_OS STREQUAL windows)
```
Comment (jkotas) [inline eng/native/configureplatform.cmake:466]: It would look better like this. Does it fix the problem that you tried to fix?

Comment (yowl) [reply eng/native/configureplatform.cmake:466]: Yes, it does thanks.

**Comment by yowl** [comment]
The failing tests are, one related to cypto which looks unrelated `Test System.Security.Cryptography.Rsa.Tests.EncryptDecrypt_TrySpan.RsaCryptRoundtrip_OaepSHA256 has failed`  and one related to wasi : 
image
`##[error].packages\microsoft.dotnet.helix.sdk\8.0.0-beta.23261.4\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(89,5): error : (NETCORE_ENGINEERING_TELEMETRY=Test) Work item System.Globalization.Tests in job 4395be70-8bf9-488a-a1ab-771ed8765dd6 has failed.`  .  That is more concerning but why it didn't fail first time is not clear.  Worth a rerun maybe?

**Review by jkotas — APPROVED** [review]

**Pull request merged**